src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -181,6 +182,14 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $opt
 							$this->nextAutoIndexes[] = $newAutoIndex;
 						}
 					}
+					if (PHP_VERSION_ID >= 80300 && count($this->keyTypes) === 1 && $offsetType->getValue() < 0) {
```

Comment:
Usually php feature detects are implemented in PhpVersion vlass

Reply:
Great!! It's exactly these kinds of comments, I need! 

I'll try and figure out how that's working! 😊